testsV2/ut_repyv2api_virtualnamespacecontextsafety.py

```diff
@@ -1,21 +1,29 @@
 """
-This unit test checks that VirtualNamespace.evaluate() forbids an unsafe context.
+Verify that VirtualNamespace.evaluate() forbids an unsafe `context`.
+
+Specifically, the `context` that a VirtualNamespace object receives 
+must be a `SafeDict`, or parseable as one. This means that the 
+context's keys must be of type `str` (which we try to violate in this 
+test), avoid particular names / prefixes / components, etc.
+
+Using an unsafe context should raise a `ContextUnsafeError` when 
+evaluatinig the VirtualNamespace.
+
+See `safe.py` for what exactly a `SafeDict` is and isn't allowed to 
+do and to contain.
 """
 
 #pragma repy 
 
-# Small code snippet
-code = "log('Bad!')\n"
-
-# Get a VN
-VN = createvirtualnamespace(code, "Bad VN")
+# Create a virtual namespace with an empty piece of code and a telling name
+vn = createvirtualnamespace("", "Bad VN")
 
-# Create a malicious context
-context = {"__metaclass__":str}
+# Create a malicious context. (Its key should be `str`, not `int`.)
+context = {123: 456}
 
 # Try to evaluate this
 try:
-  VN.evaluate(context)
+  vn.evaluate(context)
```

Comment:
Raises ContextUnsafeError as intended because [key](https://github.com/SeattleTestbed/repy_v2/pull/127/files#diff-b79f7fdd0ec68bb85e14db235284e31bR22) has to be of type `str`.